python/pyarrow/io.pxi

```diff
@@ -978,6 +1016,26 @@ def create_memory_map(path, size):
 cdef class OSFile(NativeFile):
     """
     A stream backed by a regular file descriptor.
+
+    Examples
+    --------
+    >>> import pyarrow as pa
```

Comment:
Can you add some short descriptions of what is being done? ("Creating a new file to write data to ..", "Opening a file to read..")